grails-app/services/com/unifina/service/PermissionService.groovy

```diff
@@ -147,6 +154,26 @@ class PermissionService {
 		return directPermissions
 	}
 
+	private Permission hasTransitiveDashboardPermissions(Canvas canvas, Userish userish) {
+		List<DashboardItem> items = DashboardItem.findAllByCanvas(canvas)
+		Permission permission = Permission.withCriteria(uniqueResult: true) {
```

Comment:
Will this throw if the result is not unique? I faintly remember there might be some trap like that